plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/util/ObjectStoreLocationProvider.java

```diff
@@ -40,12 +43,14 @@ public class ObjectStoreLocationProvider
     private static final int ENTROPY_DIR_DEPTH = 3;
     private final String storageLocation;
     private final String context;
+    private final boolean includePartitionPaths;
```

Comment:
Given that we don't have a dedicated table property in the Trino Iceberg connector, I'm advocating for a compatibility product tests in spark to ensure end to end that Trino does honor the `write.object-storage.partitioned-paths` table property.

The unit test is fine, but it doesn't give us full certainty that the entire write operation (and eventual read operation afterwards) succeeds through the query engine.